ksqldb-functional-tests/src/main/java/io/confluent/ksql/test/tools/TestCaseBuilder.java

```diff
@@ -52,15 +56,49 @@ private TestCaseBuilder() {
     }
 
     try {
-      final Stream<Optional<String>> formats = test.formats().isEmpty()
-          ? Stream.of(Optional.empty())
-          : test.formats().stream().map(Optional::of);
-
-      final TestLocation location = testLocator.apply(test.name());
-
-      return formats
-          .map(format -> createTest(test, originalFileName, location, format))
-          .collect(Collectors.toList());
+      final List<TestCase> testCases = new LinkedList<>();
+
+      final List<Optional<String>> formats = test.formats().isEmpty()
+          ? Collections.singletonList(Optional.empty())
+          : test.formats().stream().map(Optional::of).collect(Collectors.toList());
+
+      final List<Optional<String>> configs = test.config().isEmpty()
+          ? Collections.singletonList(Optional.empty())
+          : test.config().stream().map(Optional::of).collect(Collectors.toList());
+
+      for (final Optional<String> format : formats) {
+        for (final Optional<String> config : configs) {
+          final TestLocation location = testLocator.apply(test.name());
+
+          final Map<String, Object> updatedProperties = new HashMap<>(test.properties());
+          final Optional<String> cfg;
+          if (config.isPresent()) {
+
+            final Entry<String, Object> overwrite = Iterables.getOnlyElement(
```

Comment:
Looks like the `overwrite key` can be outside the loop and `updatedProperties` actually updated on every conf iteration

Reply:
Good catch. `overwrite` can be outside.

And yes, `updatedProperties` should be updated in each iteration, as we would set a different config each time.